### PR TITLE
Add stabilization to thermal solver + test

### PR DESCRIPTION
**Pull request by blaisb**
Add stabilization for High Peclet number in the heat transfer equation.
Adds a high-pe test demonstrating this capacity.

### CODE REVIEW

File: source/solvers/heat_transfer.cc [hunk 1]
```diff
@@ -102,7 +109,8 @@ HeatTransfer<dim>::assemble_system(
   FEValues<dim>     fe_values_ht(fe,
                              quadrature_formula,
                              update_values | update_gradients |
-                               update_quadrature_points | update_JxW_values);
+                               update_quadrature_points | update_JxW_values |
+                               update_hessians | update_JxW_values);
```
Comment (jeannejoachim) [inline source/solvers/heat_transfer.cc:113]: why is `update_JxW_values` repeated?

File: source/solvers/heat_transfer.cc [hunk 2]
```diff
@@ -217,38 +241,82 @@ HeatTransfer<dim>::assemble_system(
...
+              // Calculation of the magnitude of the velocity for the
+              // stabilization parameter
+              const double u_mag = std::max(velocity.norm(), 1e-12);
+
+              // Calculation of the GLS stabilization parameter. The
+              // stabilization parameter used is different if the simulation is
+              // steady or unsteady. In the unsteady case it includes the value
+              // of the time-step
+              const double tau =
+                is_steady(time_stepping_method) ?
```
Comment (jeannejoachim) [inline source/solvers/heat_transfer.cc:257]: I didn't know this formulation! Is it faster than a simple if/else loop?

File: source/solvers/heat_transfer.cc [hunk 3]
```diff
@@ -217,38 +241,82 @@ HeatTransfer<dim>::assemble_system(
...
+
+
 
-                          // Weak form for : - k * laplacian T + rho * cp * u *
-                          //                      gradT - f - grad(u)*grad(u) =0
+                          // Weak form for : - k * laplacian T + rho * cp *
+                          // u
+                          // *
+                          //                      gradT - f -
+                          //                      grad(u)*grad(u) =0
```
Comment (jeannejoachim) [inline source/solvers/heat_transfer.cc:296]: why is this equation on 5 lines rather than 2 as before?

**Review by jeannejoachim — APPROVED** [review]
Thank you for adding this feature so quickly! I am looking forward to see how it will impact our simulations :)

File: source/solvers/heat_transfer.cc (hunk 1 above)
Comment (blaisb) [reply source/solvers/heat_transfer.cc:113]: That's a mistake, i'll fix it.

File: source/solvers/heat_transfer.cc (hunk 3 above)
Comment (blaisb) [reply source/solvers/heat_transfer.cc:296]: Error due to automatic indentation. i'll fix that.

File: source/solvers/heat_transfer.cc (hunk 2 above)
Comment (blaisb) [reply source/solvers/heat_transfer.cc:257]: It more importantly allow you to declare a variable whose content depend on an if. This way it can be const.

**Pull request merged**